### ISSUE TITLE
Questions on experiment settings.

### ISSUE THREAD

**Issue by hw-du** [buoyancy99/diffusion-forcing#20]
Hi! Sorry to bother you again, but I am still confused about some experiment settings. I see there are some parameters in df_planning.yaml that control the episode_len, x_shape, frame_stack, and horizon, but I do not quite understand what they represent. Am I correct that episode_len controls the maximum sequence length (t) in the trajectory, and horizon controls the length of plans that the diffusion model can see during planning? And I am still not sure why x_shape is set as null, and what frame_stack controls. Thank you very much for your guidance on this question!

**Comment by buoyancy99** [comment]
`episode_len` is not a hyperparameter but some attribute provided by the environment directly, meaning how many interaction steps are allowed. The x_shape is the shape for the token. It's null because it's calculated in __init__ of the algorithm, not directly from yaml like video datasets. Frame_stack is used to reduce the number of tokens for faster training, by stacking multiple tokens into one, and is a technique commonly used in RL (e.g. stacking 4 past obs). open_loop_horizon is the number of steps to execute before mpc re-planning. e.g. You generate a plan of length of 600, take open_loop_horizon=50 steps and then replan for 550 steps.

**Comment by hw-du** [comment]
So by stacking multiple tokens into one, we are actually predicting multiple state/action pairs in a single token, instead of predicting each state/action pair step by step?

**Comment by buoyancy99** [comment]
> So by stacking multiple tokens into one, we are actually predicting multiple state/action pairs in a single token, instead of predicting each state/action pair step by step?

Yes. While DF works without this trick, training 600 token model is very slow so we almost always use frame stack

**Comment by hw-du** [comment]
Thanks! May I have another question please? For open loop horizon, "you generate a plan of length of 600, take open_loop_horizon=5- steps and then replan for 550 steps", this means only 5 steps from the generated plan are actually used. In this case, why not simply generate a plan for the next 5 steps instead of generating a plan of the whole length?

**Comment by buoyancy99** [comment]
If you only plan for short horizon like 50, how can you guide it with a goal that's 600 steps ahead? e.g. sparse reward

**Comment by hw-du** [comment]
So the whole plan generated (e.g., steps>50) can be useful for guiding diffusion denoising in the future, even if it is not directly used as the final plan, right? As a result, we can sample multiple plans at different steps, and average them for a more robust guidance instead of conditioned on a single plan generated before (this corresponds to monte carlo sampling guidance in the paper), right?

**Comment by buoyancy99** [comment]
yes. yes. Notice that you shall sample multiple to calculate average guidance gradient, not average trajectory (which is meaningless)

**Comment by hw-du** [comment]
Thank you very much!
